api/functions/process_file.py

```diff
@@ -44,36 +47,128 @@ def process_dataset_file(event: pubsub_v1.types.message, context: Context) -> No
+    upload_blob(
+        datasets_bucket_name,
+        processed_file,
+        f"{uid}/{dataset_name}/{processed_file.name}",
+    )
 
     # Check to see if we have all the files to set the dataset status as processed
     check_finished_processing(dataset_name, uid, datasets_bucket_name)
 
 
-def process_transcription_file(file_name: str, options: Dict[str, Any]) -> None:
-    pass
+def process_transcription_file(file: Path, options: Dict[str, Any]) -> Path:
+    """Transforms a downloaded transcription file into a standardised json format for training.
+
+    Parameters:
+        file: The path of the downloaded file.
+        options: The dataset processing options to apply while cleaning.
+
+    Returns:
+        The path of the processed transcription file.
+    """
+    if file.suffix == "eaf":
+        data = extract_elan_data(file, options)
+    else:
+        data = extract_text_data(file)
+
+    data = clean_data(data, options)
+
+    output_file = Path(f"/tmp/{file.stem}.json")
+    with open(output_file, "w") as data_file:
+        data_file.write(json.dumps(data))
+
+    return output_file
+
+
+def extract_elan_data(file: Path, options: Dict[str, Any]) -> List[Dict[str, str]]:
+    """Extract transcription information from an Elan file.
+
+    Parameters:
+        file: The path of the downloaded file.
+        options: The dataset processing options to apply while cleaning.
+
+    Returns:
+        A list of utterance information for the given file.
+    """
+    selection_mechanism: str = options["elanOptions"]["selectionMechanism"]
+    selection_value = options["elanOptions"]["selectionValue"]
+
+    if selection_mechanism == "tier_name":
+        return process_eaf(file, tier_name=selection_value)
+    elif selection_mechanism == "tier_type":
+        return process_eaf(file, tier_type=selection_value)
+    else:
+        try:
+            order = int(selection_value)
+        except:
+            order = 0
+
+        return process_eaf(file, tier_order=order)
+
+
+def extract_text_data(file: Path) -> List[Dict[str, str]]:
+    """Extract transcription information from a text file.
+
+    Parameters:
+        file_name: The name of the downloaded file.
+
+    Returns:
+        A list of utterance information for the given file.
+    """
+    with open(file) as transcription_file:
+        transcription = transcription_file.read()
+
+    # Returning a dummy format without start and end times.
+    return [
+        {
+            "audio_file_name": file.stem + ".wav",
+            "transcript": transcription,
+        }
+    ]
+
+
+def clean_data(
+    transcription_data: List[Dict[str, str]], options: Dict[str, Any]
+) -> List[Dict[str, str]]:
+    """Takes a list of utterance information and cleans it based off the provided
+    dataset processing options.
+
+    Parameters:
+        transcription_data: A list of information about utternaces in the transcription.
```

Comment:
`utternances` - typo, should be `utterances`.

Reply:
Good catch. Fixed in next revision